DEV_GUIDE.md

```diff
@@ -340,10 +341,12 @@ has been applied ineffectively.
 * `SKIP_TEARDOWN`: variable for development purposes to avoid keep deploying and deleting deployments each run. Default value: `false`
 * `CONTAINER_CONFIG_PATH`: directory where `config.json` file is located. This file contains the pull secrets to be used by
 the container engine. Default value: `$HOME/.docker/config.json`
-* `VAULT_CHART_VERSION`: version of Vault Helm Chart to be used by the System Tests for Envelope Encryption. Default value: `0.27.0`
+* `VAULT_CHART_VERSION`: version of Vault Helm Chart to be used by the System Tests for Record Encryption. Default value: `0.27.0`
 * `SKIP_STRIMZI_INSTALL`: skip strimzi installation. Default value: `false`
 * `KAFKA_CLIENT`: client used to produce/consume messages. Default value: `strimzi_test_client`. Currently supported values: `strimzi_test_client`, `kaf`, `kcat`
-
+* `AWS_ACCESS_KEY_ID`: key id of the aws account to be used for KMS. Localstack will be used by default. Default value: `test`
```

Comment:
We need to split these into a pair of access/secret keys. One set for the KMS itself and one set for the TestFacade.  This will give us the ability to verify that the KMS implementation does not expect excess privileges.

```
AWS_KMS_ACCESS_KEY_ID
AWS_KMS_SECRET_ACCESS_KEY
```

```
AWS_TEST_FACADE_KMS_ACCESS_KEY_ID
AWS_TEST_FACADE_KMS_SECRET_ACCESS_KEY
```

I would make the switch between AWS and LocalStack explicit too.

AWS_USE_CLOUD=(true|false)

Reply:
I don't see the need of having two different key pairs, if in the end both `Localstack` and AWS cloud need the env variable `AWS_ACCESS_KEY_ID` to be set. If we split the pairs, when somebody external to the implementation tries to run the tests, how do you explain what is a Test Facade? I'm always thinking of keep simple the test execution.

I can see adding a new env variable like `AWS_USE_CLOUD` could be enough to assure that we are not using AWS cloud when it is not needed.

Reply:
The point @k-wall is making is that the `AWS_ACCESS_KEY_ID` should not have permissions to create keys in AWS, that permission should reserved for `AWS_TEST_FACADE_KMS_ACCESS_KEY_ID` because its not something the actual implementation should be able to do. 

While I hear @franvila's point about complicating execution I think we should use independent credentials as a good example for everyone reading the code. 

I don't see the `TEST_FACADE` naming as being helpful to users either so maybe `KMS_TEST_ADMIN_*` or `TEST_KMS_ADMIN_*` naming (we can throw a AWS there somewhere as well if desired)?

Reply:
> The point @k-wall is making is that the `AWS_ACCESS_KEY_ID` should not have permissions to create keys in AWS

Sorry, but I still don't understand this point. The name of the environmental variable needed by AWS to run anything is `AWS_ACCESS_KEY_ID`. If you don't set this variable, nothing works. The content of the variable could be something related to what is on kroxylicious [documentation](https://kroxylicious.io/docs/v0.6.0/#filter-actor-2)  with the corresponding policy restrictions.

As an AWS user, I can see easier to understand that I have to set `AWS_ACCESS_KEY_ID`, what I need in an usual configuration, than `AWS_TEST_FACADE_KMS_ACCESS_KEY_ID`.

Reply:
I think I got it!

- Use `kroxylicious` user's access key id in the config map when deploying kroxylicious to only read the kek to encrypt/decrypt the record
- Use `kroxylicious-admin` user's access key id as the usual `AWS_ACCESS_KEY_ID` for the facade to create/delete keks into the AWS Cloud.

In that case, I would keep `AWS_ACCESS_KEY_ID` for the facade, as it is the main access key we need for testing purposes and it is the name of the environmental variable used on our CLI to access to the AWS Cloud. Then create a new one for Kroxylicious ConfigMap `AWS_KROXYLICIOUS_ACCESS_KEY_ID`, as the user-name in fact is `kroxylicious`

Reply:
Yes that's the idea we are trying to express.

I'm ambivalent about the naming and which one gets the "obvious" name the important point is the split. I would generally give things that are only needed for testing purposes a more explicit name to make clear that it's not part of the production usage. However the main users of these variables are people setting up and executing tests so maybe it makes more sense to do it from their perspective...